### PR TITLE
Netfx sample fixes

### PR DESCRIPTION
**Pull request by brandonh-msft**
fixes to resolve building warnings and runtime issues

cc @cleemullins @tomlm

### CODE REVIEW

File: samples/Microsoft.Bot.Samples.EchoBot-AspNet461/App_Start/WebApiConfig.cs [hunk 1]
```diff
@@ -38,10 +38,11 @@ public static void Register(HttpConfiguration config)
...
             UnityConfig.Container.RegisterSingleton<BotFrameworkAdapter>(
                 new Unity.Injection.InjectionConstructor(
-                    new BotFrameworkAdapter(ConfigurationManager.AppSettings[@"MicrosoftAppId"], ConfigurationManager.AppSettings[@"MicrosoftAppPassword"])
-                        .Use(new BotStateManager(new MemoryStorage()))
+                    ConfigurationManager.AppSettings[@"MicrosoftAppId"],
+                    ConfigurationManager.AppSettings[@"MicrosoftAppPassword"])
                 )
-            );
+            .Resolve<BotFrameworkAdapter>()
+            .Use(new BotStateManager(new MemoryStorage()));
```
Comment (cleemullins) [inline samples/Microsoft.Bot.Samples.EchoBot-AspNet461/App_Start/WebApiConfig.cs]: Note: This is no longer correct. The BotStateManager is gone, in place of the UserStateManagerMiddleware and the ConversationStateManagerMiddleware.

Comment (brandonh-msft) [reply samples/Microsoft.Bot.Samples.EchoBot-AspNet461/App_Start/WebApiConfig.cs]: pullin' the rug out from under me! ;)

**Comment by brandonh-msft** [comment]
⏲

**Comment by brandonh-msft** [comment]
@cleemullins please pull this in. i'm having to continually rebase while it sits here, and the sample is broken w/o these fixes.

**Comment by cleemullins** [comment]
Thanks Brandon!

**Pull request merged**